eWonConnector-gateway/src/main/java/com/hms_networks/americas/sc/ignition/data/tagwrite/BufferedTagWriteManager.java

```diff
@@ -0,0 +1,432 @@
+  }
+
+  /**
+   * Processes a buffered tag write using the specified parameters. This method will add the tag
+   * write to the buffered tag write queue, and schedule the buffered tag write processing thread to
+   * process the queue.
+   *
+   * @param managedTagProvider the managed tag provider where the tag is located
+   * @param connectorSettings the connector settings
+   * @param m2WebEwon the M2Web Ewon object for the Ewon where the tag is located
+   * @param m2WebEwonEBDTag the M2Web Ewon EBD tag object for the tag to write
+   * @param tagPath the managed tag provider tag path
+   * @param tagValue the tag value to write
+   * @since 1.0.0
+   */
+  private static void writeTagBuffered(
+      ManagedTagProvider managedTagProvider,
+      EwonConnectorSettings connectorSettings,
+      M2WebEwon m2WebEwon,
+      M2WebEwonEBDTag m2WebEwonEBDTag,
+      TagPath tagPath,
+      Object tagValue) {
+
+    // Add buffered tag write to queue and schedule processing
+    synchronized (BUFFERED_TAG_WRITE_QUEUE) {
+      // Add buffered tag write to queue
+      BUFFERED_TAG_WRITE_QUEUE.add(
+          new BufferedTagWrite(m2WebEwon, m2WebEwonEBDTag, tagPath, tagValue));
+
+      // Schedule buffered tag write processing
+      scheduleBufferedWriteProcessing(managedTagProvider, connectorSettings);
+    }
+
+    // Log debug message
+    if (connectorSettings.isDebugEnabled()) {
+      LOGGER.debug(
+          "Tag write scheduled (buffered) for tag [{}] with value [{}].",
+          tagPath.toString(),
+          tagValue.toString());
+    }
+  }
+
+  /**
+   * Schedules the buffered tag write processing thread to process the buffered tag write queue.
+   * This method will only schedule the processing thread if it is not already running. If the
+   * processing thread is already running, this method will do nothing, as the processing thread
+   * will process the updated queue automatically.
+   *
+   * @param managedTagProvider the managed tag provider where the tag is located
+   * @param connectorSettings the connector settings
+   * @since 1.0.0
+   */
+  private static synchronized void scheduleBufferedWriteProcessing(
+      ManagedTagProvider managedTagProvider, EwonConnectorSettings connectorSettings) {
+    // If buffered tag write processing thread is not running, create and start it
+    if (bufferedTagWriteProcessingThread == null || !bufferedTagWriteProcessingThread.isAlive()) {
+      // Create new thread
+      bufferedTagWriteProcessingThread =
+          new Thread(
+              () -> {
+                // Create boolean to track if thread should continue running
+                boolean continueRunning = true;
+
+                // Loop until queue is empty
+                long totalBufferedWriteCount = 0;
+                while (continueRunning) {
+                  // Wait for buffer length
+                  try {
+                    Thread.sleep(writeBufferLengthMilliseconds);
```

Comment:
I think this is fine for now, a better choice would be to await notification that Buffered_tag_write_queue is not empty

Reply:
That would change the core functionality of this feature. This feature is intended to buffer tag writes within a configurable period of time. 

- If there are tag writes buffered while this thread is already running, it will set `continueRunning` to true and loop again. 
- If there is no tag writes buffered, `continueRunning` is set to false, and the thread is shutdown.
  - When the next buffered write is received, a new thread is created as/if necessary.